gobblin-core/src/main/java/gobblin/publisher/TimestampDataPublisher.java

```diff
@@ -32,10 +33,51 @@ public TimestampDataPublisher(State state) throws IOException {
     timestamp = String.valueOf(System.currentTimeMillis());
   }
 
+  /**
+   * Make sure directory exists before running {@link BaseDataPublisher#publishData(WorkUnitState, int, boolean, Set)}
+   * so that tables will be moved one at a time rather than all at once
+   */
+  @Override
+  protected void publishData(WorkUnitState state, int branchId, boolean publishSingleTaskData,
+      Set<Path> writerOutputPathsMoved) throws IOException {
+    Path publisherOutputDir = getPublisherOutputDir(state, branchId);
+    if (!this.publisherFileSystemByBranches.get(branchId).exists(publisherOutputDir)) {
+      WriterUtils.mkdirsWithRecursivePermission(this.publisherFileSystemByBranches.get(branchId),
+          publisherOutputDir, this.permissions.get(branchId));
+    }
+    super.publishData(state, branchId, publishSingleTaskData, writerOutputPathsMoved);
+  }
+
+  /**
+   * Update destination path to put db and table name in format "dbname.tablename" using {@link #getDbTableName(String)}
+   * and include timestamp
+   *
+   * Input dst format: {finaldir}/{schemaName}
+   * Output dst format: {finaldir}/{dbname.tablename}/{currenttimestamp}
+   */
   @Override
-  protected Path getPublisherOutputDir(WorkUnitState workUnitState, int branchId) {
-    Path outputDir = super.getPublisherOutputDir(workUnitState, branchId);
-    outputDir = new Path(outputDir, timestamp);
-    return outputDir;
+  protected void movePath(ParallelRunner parallelRunner, State state, Path src, Path dst, int branchId)
+      throws IOException {
+
+    String outputDir = dst.toString().substring(0, dst.toString().lastIndexOf('/'));
```

Comment:
You can use `dst.getName()` and `dst.getParent().getName()` to replace the parsing of `/` here.